ios/FluentUI/Navigation/NavigationBar.swift

```diff
@@ -727,12 +853,37 @@ open class NavigationBar: UINavigationBar, TokenizedControlInternal {
     private func needsShadow(for navigationItem: UINavigationItem?) -> Bool {
         switch navigationItem?.navigationBarShadow ?? .automatic {
         case .automatic:
-            return !showsLargeTitle && style == .system && navigationItem?.accessoryView == nil
+            return !usesLeadingTitle && style == .system && !systemWantsCompactNavigationBar && navigationItem?.accessoryView == nil
         case .alwaysHidden:
             return false
         }
     }
 
+    private func updateSubtitleView(for navigationItem: UINavigationItem?) {
+        guard let navigationItem = navigationItem, !usesLeadingTitle else {
+            // Use the default title view
+            navigationItem?.titleView = nil
+            return
+        }
+
+        let customTitleView = TwoLineTitleView(style: style == .primary ? .primary : .system)
+        customTitleView.tokenSet.setOverrides(from: tokenSet, mapping: [
+            .titleColor: .titleColor,
+            .titleFont: .titleFont,
+            .subtitleColor: .subtitleColor,
+            .subtitleFont: .subtitleFont
+        ])
+        customTitleView.setup(navigationItem: navigationItem)
+        if navigationItem.titleAccessory == nil {
+            // Use default behavior of requesting an accessory expansion
+            customTitleView.delegate = self
+        }
+
+        // For some strange reason, embedding the TwoLineTitleView inside a UIStackView
```

Comment:
nit: remove "For some strange reason" as it doesn't really add any useful extra info to the reader

Reply:
Reworded.